api/v1beta1/agentserviceconfig_types.go

```diff
@@ -122,6 +122,23 @@ type AgentServiceConfigSpec struct {
 	// +optional
 	OSImageAdditionalParamsRef *corev1.LocalObjectReference `json:"OSImageAdditionalParamsRef,omitempty"`
 
+	// AssistedServiceIngressHost is the hostname to be assigned to the assisted-service ingress.
+	// Has no effect when running on an OpenShift cluster.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Assisted Service Ingress hostname"
+	// +optional
+	AssistedServiceIngressHost string `json:"assistedServiceIngressHost,omitempty"`
+
+	// ImageServiceIngressHost is the hostname to be assigned to the assisted-image-service ingress.
+	// Has no effect when running on an OpenShift cluster.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Assisted Image Service Ingress hostname"
+	// +optional
+	ImageServiceIngressHost string `json:"imageServiceIngressHost,omitempty"`
+
+	// IngressClassName is the name of the ingress class to be used when configuring ingress resources.
+	// Has no effect when running on an OpenShift cluster.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Ingress Class Name"
+	// +optional
+	IngressClassName string `json:"ingressClassName,omitempty"`
```

Comment:
Does anyone have an issue with adding these to the top level of agentserviceconfig?
Should we nest them in some kind of kube-specific type?

@avishayt @mhrivnak any thoughts?

Reply:
Generally this looks like a good opportunity to nest them. Then it's clear that the fields together represent a specific feature. You either create the nested struct and fill it out appropriately, or you are free to ignore the whole thing. Validation is a bit simpler that way too, especially for inter-field dependencies.

Reply:
I agree, nested

Reply:
How about `IngressConfig` as the field and type name with fields `ClassName`, `ImageServiceHostname`, and `AssistedServiceHostname`?

Reply:
Are any of these fields more of a "config" than everything else in the Spec? Assuming not, I'd default to `Ingress.ClassName`, `Ingress.ImageServiceHostname`, etc.